bitcoin/src/address/mod.rs

```diff
@@ -319,11 +318,8 @@ impl From<Network> for KnownHrp {
 /// * [BIP341 - Taproot: SegWit version 1 spending rules](https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki)
 /// * [BIP350 - Bech32m format for v1+ witness addresses](https://github.com/bitcoin/bips/blob/master/bip-0350.mediawiki)
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-///
-/// The `#[repr(transparent)]` attribute is used to guarantee that the layout of the
-/// `Address` struct is the same as the layout of the `AddressInner` struct. This attribute is
-/// an implementation detail and users should not rely on it in their code.
-///
+// The `#[repr(transparent)]` attribute is used to guarantee the layout of the `Address` struct. It
+// is an implementation detail and users should not rely on it in their code.
```

Comment:
In 9cabb291e722fe86224b27bebd695a770d7a6a5d:

This needs to be a public doccomment, since `#[repr(transparent)]` is visible in the public docs.

Reply:
Oh is it, I'll look again, I checked yesterday before doing this, my bad.

Reply:
How is it visible in the docs? I built and read the file and I don't see any mention of `repr(transparent)`?

Reply:
Oh, huh, I guess I'm wrong. This is surprising to me given that the `repr` is a user-visible thing (at least, it is when the inner type is public).

Reply:
This must be some new change since last time I looked it did show up. But it's a very nice thing since this was an unresolved issue for a while and we can change it in `Script` and `PushBytes` as well.